docs/specifications/tests/DNSSEC-TP/dnssec17.md

```diff
@@ -106,40 +115,53 @@ DS17_MIXED_DELETE_CDNSKEY            | ERROR   | "Delete" CDNSKEY record is mixe
 
 7.  For each name server IP in the *CDNSKEY RRsets* set do:
 
-    1. If the CDNSKEY RRset is non-empty, get the RRset and its RRSIG
-       records, else go to next name server IP address. The set of the RRSIG
-       records may be empty.
-    2. If any CDNSKEY record is a "delete" CDNSKEY, then do:
+    1. If the CDNSKEY RRset is empty go to next name server IP address.
+    2. Get the CDNSKEY RRset and the associated RRSIG records, if any, from the
+       *CDNSKEY RRsets* set for the name server IP.
+    3. If any CDNSKEY record is a "delete" CDNSKEY, then do:
        1. If there is more than a single CDNSKEY record then add the
           name server IP to the *Mixed Delete CDNSKEY* set.
        2. Else, add the name server IP address to the *Delete CDNSKEY*
           set.
-       3. Go to next name server IP.
-    3. Get the DNSKEY and its RRSIG records from the *DNSKEY RRsets* for the same
-       name server IP. The set of RRSIG records may be empty.
-    4. If there are no DNSKEY records, then do:
+    4. Get the DNSKEY RRset and the associated RRSIG records, if any, from the
+       *DNSKEY RRsets* for the same name server IP.
+    5. If there are no DNSKEY records, then do:
        1. Add name server IP address to the *No DNSKEY RRset* set
           (duplicates not possible).
        2. Go to next name server IP.
-    5. Repeat the following steps for each CDNSKEY record:
-       1. Compare the CDNSKEY record with the DNSKEY records.
-       2. If the CDNSKEY record does not match any DNSKEY record then
-          add the name server IP address and the key tag derived from
-          the CDNSKEY record to the *No Match CDNSKEY With DNSKEY* set.
-       3. Else, if there is no RRSIG for the DNSKEY RRset created by
-          the DNSKEY record matching the CDNSKEY record then add the
-          name server IP address and key tag of CDNSKEY record to the
-          *DNSKEY Not Signed By CDNSKEY* set.
-    6. If there are no RRSIG records for the CDNSKEY RRset, then add
-       the name server IP address to the *CDNSKEY Not Signed* set.
-    7. Else, for each RRSIG (CDNSKEY) do:
-       1. If the key tag of the RRSIG does not match any DNSKEY then
-          add the name server IP address and key tag to the
+    6. Repeat the following steps for each CDNSKEY record unless it is a "delete"
+       CDNSKEY record:
+       1. If bit 7 of the flags field of the CDNSKEY record is unset (value 0)
+          then add the name server IP address and the key tag derived from
```

Comment:
Does the word "derived" here (and twice more below) refer to the key tag calculation described in [RFC 4034 Appendix B](https://datatracker.ietf.org/doc/html/rfc4034#appendix-B)? In that case maybe "calculated" would be a more helpful word?

Reply:
Good point. I'll update.